contracts/protocol/bases/mixins/FundsManager.sol

```diff
@@ -182,11 +182,17 @@ contract FundsManager {
      * @param _tokenAddress - address of the token to be transferred
      */
     function checkTrustedForwarder(address _tokenAddress) internal view virtual returns (bool) {
-        try ERC2771Context(_tokenAddress).trustedForwarder() returns (address trustedForwarder) {
-            return trustedForwarder == address(this);
-        } catch {
-            return false;
+        (bool success, bytes memory returnData) = _tokenAddress.staticcall(
+            abi.encodeCall(ERC2771Context.trustedForwarder, ())
+        );
+
+        if (success) {
+            if (returnData.length != SLOT_SIZE) {
+                return false;
+            }
+            return abi.decode(returnData, (address)) == address(this);
```

Comment:
I don't think this decodes gracefully if the returned data is for example uin256 with value > 2^160.

Probably sth like
`address(uint160(abi.decode(returnData, (uint256)))) == address(this);`
is more robust.